### PR TITLE
My first couple of months at Codurance

### PR DESCRIPTION
**Pull request by neomatrix369**
Looking forward to your constructive comments.

I couldn't stop myself from writing more, but I wanted it to be a concise post. So if you think I missed some aspect or I could reword something better, I'll be happy to receive your feedback.

Thanks in advance.

### CODE REVIEW

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+image:
+    src: "/assets/img/custom/blog/Five\ Characteristics\ of\ a\ Great\ Company\ Culture.jpg"
+    attribution:
+        text: Together we are a formidable force
+        href: http://www.success.com/sites/default/files/main/blogposts/Five%20Characteristics%20of%20a%20Great%20Company%20Culture.jpg
+---
+
+### Some background
+
+Some of you may know me from the various meetups in the city, especially my attendance at a number of [LJC](www.meetup.com/Londonjavacommunity/) and [LSCC](www.meetup.com/london-software-craftsmanship) meetup events. Attending these events I learnt about various conferences like **Devoxx**, **SoCraTes**, **JAX London**, **Java2Days**, **OpenFest**, and some others, and have had the good fortune at being at such events and even presenting at some of them. My subject was usually around [Adopt OpenJDK](https://java.net/projects/adoptopenjdk/pages/AdoptOpenJDK) and [Adopt-a-JSR](https://java.net/projects/adoptajsr/pages/Home) whilst wearing my [LJC](www.meetup.com/Londonjavacommunity/) Advocate hat, but I also took the chance to present on other topics.
```
Comment (11PJ11) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md]: good fortune of being

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+I also noticed that many things I learnt at such events and conferences wouldn't always immediately recognised or accepted at the workplace. And moving to another work environment didn't always solve this problem fully. I found that I wasn't learning and applying what I wanted to from my peers and very few were really in tuned with what the community was about. So one fine day I decided to take charge of my career and make a serious decision and take up the Apprenticeship program offered by [Codurance](http://codurance.com/) and go through the process.
```
Comment (dhatanian) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md]: wouldn't always **be** immediately recognised

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+I was urged to go this way after being inspired by [Sandro](https://twitter.com/sandromancuso)'s book: [The Software Craftsman](http://www.amazon.co.uk/books/dp/0134052501), attending all the [SoCraTes UK](http://socratesuk.org/) conferences, and meeting with developers who valued and took pride of their work namely their craft. 
+
+### Where we are just now
+
+It's now been nearly two months since I have been working for 'Codurance', a formidable force. And so its also about time that I share my experiences with my fellow mates and the community around me.
+
+During my first few weeks at [Codurance](http://codurance.com/), I have been busy learning various things that have been chalked out for becoming a craftsman.
```
Comment (dhatanian) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md:39]: You should probably put the link to Codurance's website on the first mention of the company. No need to put it twice.

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+I also noticed that many things I learnt at such events and conferences wouldn't always immediately recognised or accepted at the workplace. And moving to another work environment didn't always solve this problem fully. I found that I wasn't learning and applying what I wanted to from my peers and very few were really in tuned with what the community was about. So one fine day I decided to take charge of my career and make a serious decision and take up the Apprenticeship program offered by [Codurance](http://codurance.com/) and go through the process.
+
+I was urged to go this way after being inspired by [Sandro](https://twitter.com/sandromancuso)'s book: [The Software Craftsman](http://www.amazon.co.uk/books/dp/0134052501), attending all the [SoCraTes UK](http://socratesuk.org/) conferences, and meeting with developers who valued and took pride of their work namely their craft.
```
Comment (dhatanian) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md]: You have already linked to Sandro's Twitter before, no need to do it twice

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md (hunk 3 above)
Comment (dhatanian) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md:39]: chalked out by whom ? did you decide what you needed/wanted to learn or was that coming from your mentor ?

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+I found our office environment to be conducive to learning, sharing and collaboration. We even have a pairing rota that we use from time-to-time to record or suggest pairing sessions during the week.
+
+Just last week we did a group code review and time-boxed ourselves, performed a retrospective at the end of each interval and ensured we delivered a good chunk of the feedback before close of play.
+
+The apprentice meetings are full of fun -- we are accompanied by at least one craftsperson, who disperses their knowledge and experience from a wide variety of topics designed to help us in the journey and fill the gaps in our knowledge and experience.
+
+
+### Socialising and sharing
+
+We share links to events, conferences, tweets, interesting articles, videos, blog posts, etc... via slack, document discussions and brain dumps via Google doc, huddles during lunch- and tea- breaks to talk about anything we are working on. Thanks to the library of printed and digital books to our disposal, the huge collection of [blogs](http://codurance.com/blog) and [videos](http://codurance.com/videos/) on our site.
```
Comment (dhatanian) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md]: huge collection of blogs --> huge collection of blog posts

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+Just last week we did a group code review and time-boxed ourselves, performed a retrospective at the end of each interval and ensured we delivered a good chunk of the feedback before close of play.
+
+The apprentice meetings are full of fun -- we are accompanied by at least one craftsperson, who disperses their knowledge and experience from a wide variety of topics designed to help us in the journey and fill the gaps in our knowledge and experience.
+
+
+### Socialising and sharing
+
+We share links to events, conferences, tweets, interesting articles, videos, blog posts, etc... via slack, document discussions and brain dumps via Google doc, huddles during lunch- and tea- breaks to talk about anything we are working on. Thanks to the library of printed and digital books to our disposal, the huge collection of [blogs](http://codurance.com/blog) and [videos](http://codurance.com/videos/) on our site. 
+
+A bi-weekly catchup involves sharing of knowledge via lightening talks, discussions and pairing sessions on pet projects over pizzas and beer (and of course veggies and non-alcoholic beverages for the [teetotalers](https://en.wikipedia.org/wiki/Teetotalism)).
```
Comment (dhatanian) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md]: lightening --> lightning

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+The apprentices have also collective started a Flying Fridays which of course happens on every Friday, sometimes its dinner at a nearby restaurant, whiles at other times an indoor movie over snacks and drinks in our office premises.
```
Comment (dhatanian) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md]: collective --> collectively
sometimes its dinner --> sometimes it's dinner
whiles at other times --> while at other times

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md (hunk 2 above)
Comment (neomatrix369) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md]: done thnx

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md (hunk 3 above)
Comment (neomatrix369) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md:39]: done thnx

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md (hunk 5 above)
Comment (neomatrix369) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md]: done thnx

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md (hunk 6 above)
Comment (neomatrix369) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md]: done thnx

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md (hunk 7 above)
Comment (neomatrix369) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md]: done thnx

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+During this time I met a lot of people with various levels of experience and my interest and urge to learn more about the Java/JVM platform, Code Quality, Software Design, XP Practices, Software Craftsmanship, etc..., were on the rise and saw no end. And whilst attending these events I came across [Sandro](https://twitter.com/sandromancuso) and [Mash](https://twitter.com/mashooq), who were in those days hosting [LSCC](www.meetup.com/london-software-craftsmanship) events. I went to many of [LSCC](www.meetup.com/london-software-craftsmanship) events, especially liked the hands-on sessions (which are still my favourite). These events always took place in the evenings and sometimes during weekends, and in most instances organised at [Skillsmatter](https://skillsmatter.com/).
+
```
Comment (sandromancuso) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md:29]: Remove the last line. "These events always... " It's irrelevant information and takes the reader's attention away from the blog.

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+I also noticed that many things I learnt at such events and conferences wouldn't always be immediately recognised or accepted at the workplace. And moving to another work environment didn't always solve this problem fully. I found that I wasn't learning and applying what I wanted to, from my peers and very few were really in tuned with what the community was about. So one fine day I decided to take charge of my career and make a serious decision and take up the Apprenticeship program offered by [Codurance](http://codurance.com/) and go through the process.
+
```
Comment (sandromancuso) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md:31]: The sentence "I found that I wasn't ...., from my peers and very few... " needs a bit of rewriting.

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+- community
+- mentor
+- mentee
+- mentoring
+---
+
+### Some background
+
+Some of you may know me from the various meetups in the city, especially my attendance at a number of [LJC](www.meetup.com/Londonjavacommunity/) and [LSCC](www.meetup.com/london-software-craftsmanship) meetup events. Attending these events I learnt about various conferences like **Devoxx**, **SoCraTes**, **JAX London**, **Java2Days**, **OpenFest**, and some others, and have had the good fortune of being at such events and even presenting at some of them. My subject was usually around [Adopt OpenJDK](https://java.net/projects/adoptopenjdk/pages/AdoptOpenJDK) whilst wearing my [LJC](www.meetup.com/Londonjavacommunity/) Advocate hat, but I also took the chance to present on other topics.
+
```
Comment (sandromancuso) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md:27]: From "Attending these events... " until the end of the paragraph. It's a bit too verbose and bumpy. You could replace it with something like "At these community events, I learned about some interesting conferences like Devoxx..., which I ended up attending and later presenting on various topics including the Adopt OpenJDK."

Remove the thing about wearing a LJC advocate hat.

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+
+### Where we are just now
+
+It's now been nearly two months since I have been working for [Codurance](http://codurance.com/), a formidable force. And so it's also about time that I share my experiences with my fellow mates and the community around me.
+
+During my first few weeks at [Codurance](http://codurance.com/), I have been busy learning various things that have been chalked out for becoming a craftsman.
+
+When working on a kata or learning a concept, we paired or did what is known as 'mob programming' along with other apprentices and craftspersons. And most of the time used the [pomodoro technique](https://en.wikipedia.org/wiki/Pomodoro_Technique). Time boxing our work in intervals is something done both in groups and working individually. We would have a lot of discussions and retrospectives after working on a problem or writing some code from scratch. 
+
```
Comment (sandromancuso) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md:42]: We use "craftsmen" as the collective. Craftspersons is a tongue twisting and it highlights that you are trying to be politically correct, but normally it has a negative effect (using unnatural language).

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+### Structure of my program
+
+We used an internal tool based on the concept of [Impact Mapping](http://www.impactmapping.org/). I soon got interested in it when I saw my colleague [Franzi](https://twitter.com/Singsalad) (who is now a craftswoman) had used it to plan out her Apprenticeship route. Such a tool helps map out our goals and the tasks we need to perform to achieve it. And this can differ from person-to-person, depending on what they want to work on (driven by the Apprentice).
+
+My mentor and other craftspersons reviewed them to get an idea of what I wanted to achieve for myself. And then its upto me to apply my own drive and perseverance to achieve the individual stories. I and my mentor meet and talk informally on a regular basis, many a times pairing on a kata or a project or on the white board trying to get my head around a concept.
+
```
Comment (sandromancuso) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md:48]: You can say I'm your mentor if you want. :) 

I and my mentor => My mentor and I

"... many a times.. " => many times

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+### Days in the life of an Apprentice
+
+I found the working hours quite flexible, remote working is also an optio as we can do given everything is on the cloud these days. Our co-founders are understanding and compassionate about our individual situations.
```
Comment (sandromancuso) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md]: Remote working is not necessary an option. It's an option when you are on the bench or if the client allows, if you are in a project. I don't want to send the wrong message to people that may want to join Codurance in the future. 

Remove "everything is on the cloud these days". Everyone knows that and the information is redundant.

File: _posts/2015-11-23-my-first-couple-of-months-at-codurance.md [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+
+We share links to events, conferences, tweets, interesting articles, videos, blog posts, etc... via slack, document discussions and brain dumps via Google doc, huddles during lunch- and tea- breaks to talk about anything we are working on. Thanks to the library of printed and digital books to our disposal, the huge collection of [blog posts](http://codurance.com/blog) and [videos](http://codurance.com/videos/) on our site. 
+
+The apprentices and some craftspersons have collectively started social event which of course happens every Friday, sometimes it's dinner at a nearby restaurant, while at other times an indoor movie over snacks and drinks in our office premises.
+
+### It is worthwhile and that's why we are here
+
+It is a privilege to be able to work alongside very experienced craftspersons from our industry. We are very lucky to have the opportunity to be guided and mentored by talented and like minded developers.
+
+This is one of my first jobs where the company has a completely flat hierarchy and where we share similar values amongst ourselves.
```
Comment (sandromancuso) [inline _posts/2015-11-23-my-first-couple-of-months-at-codurance.md]: "This is one of my first jobs..." is not so clear.

**Pull request merged**